src/components/board/board.ts

```diff
@@ -89,6 +93,17 @@ export interface Board extends Resource {
     digitalInterruptName: string,
     extra?: StructType
   ): Promise<number>;
+  /**
+   * Stream digital interrupt ticks on the board.
+   *
+   * @param interrupts - Names of the interrupts to stream.
+   * @param queue - Array to put the ticks in.
+   */
+  streamTicks(
+    interrupts: string[],
+    queue: Tick[],
```

Comment:
A digital interrupt tick is triggered when the gpio pin changes from high to low or vice versa, the tick struct contains the name of the pin, if its high or low, and the time that the tick occurred. The ticks have to be sent from the board to the server to the client. Its a parameter rather than being created in the function so that whatever is calling this client function can provide the array, the client then adds the ticks to it when it receives them from the server, and then the caller can get the ticks added to the array and use them however it needs to.

Reply:
Thanks for the explanation!